config/nrfconnect/chip-module/CMakeLists.txt

```diff
@@ -319,6 +324,10 @@ if (CONFIG_CHIP_EXAMPLE_DEVICE_INFO_PROVIDER)
     target_include_directories(chip INTERFACE ${CHIP_ROOT}/examples/providers)
 endif()
 
+if (CONFIG_CHIP_FACTORY_DATA)
+    include(${CHIP_ROOT}/src/platform/nrfconnect/factory_data/CMakeLists.txt)
```

Comment:
I think that if we want to put the factory data provider in `src/platform` it should be built using GN. You can add another GN argument similarly to `chip_malloc_sys_heap`. Alternatively, we can put it in `examples/platform/nrfconnnect` and include explicitly in example's CMakeLists.txt.

Reply:
As discussed F2F, I managed to compile it in the src/platform using GN.